File: app/src/main/java/com/ysxsoft/imtalk/chatroom/utils/SpUtils.java
package com.ysxsoft.imtalk.chatroom.utils;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Create By 胡
 * on 2019/8/5 0005
 */
public class SpUtils {

    public static void saveSp(Context context, String key, String value) {
        SharedPreferences.Editor sp = context.getSharedPreferences("SAVE", Context.MODE_PRIVATE).edit();
        sp.putString(key, value);
        sp.apply();
    }

    public static String getSp(Context context, String key) {
        SharedPreferences sp = context.getSharedPreferences("SAVE", Context.MODE_PRIVATE);
        return sp.getString(key, "");
    }

    public static void deleteSp(Context context, String key, String value) {
        SharedPreferences.Editor deleteSp = context.getSharedPreferences("SAVE", Context.MODE_PRIVATE).edit();
        deleteSp.clear();
        deleteSp.apply();
    }


}

File: app/src/main/java/com/ysxsoft/imtalk/utils/SpUtils.kt
package com.ysxsoft.imtalk.utils

import android.content.Context

object SpUtils {
    fun saveSp(context: Context, key: String, value: String) {
        val sp = context.getSharedPreferences("SAVE", Context.MODE_PRIVATE).edit()
        sp.putString(key, value)
        sp.commit()
    }

    fun getSp(context: Context, key: String): String {
        val getSp = context.getSharedPreferences("SAVE", Context.MODE_PRIVATE)
        return getSp.getString(key, "")
    }

    fun deleteSp(context: Context) {
        val deleteSp = context.getSharedPreferences("SAVE", Context.MODE_PRIVATE).edit()
        deleteSp.clear()
        deleteSp.commit()
    }

    fun savePwd(context: Context, key: String, value: String) {
        val sp = context.getSharedPreferences("PWD", Context.MODE_PRIVATE).edit()
        sp.putString(key, value)
        sp.commit()
    }

    fun getPwd(context: Context, key: String): String {
        val getSp = context.getSharedPreferences("PWD", Context.MODE_PRIVATE)
        return getSp.getString(key, "")
    }

    fun deletePWD(context: Context) {
        val deleteSp = context.getSharedPreferences("PWD", Context.MODE_PRIVATE).edit()
        deleteSp.clear()
        deleteSp.commit()
    }
}
